### ISSUE TITLE
Globalize("en").parseNumber("35NaN") results in 35 - not NaN

### ISSUE THREAD

**Issue by robwade** [jquery/globalize#632]
Results from IE and FireFox consoles:
> window.Globalize("en").parseNumber("35NaN")
> << 35

To fix, add this code the bottom of Number.js
> // Check for invalid prefixes and suffixes
> if ((prefix && prefix !== negativePrefix) || (suffix && suffix !== negativeSuffix)) {
>     return NaN;
> }
prior to:
> return number;
in the:
> numberParse function
of:
> number.js

Verision Details:
>	Globalize v1.1.1
>	Date: 2016-02-04T12:01Z
>	CLDR: {
>	  "main": {
>	    "en": {
>	      "identity": {
>	        "version": {
>	          "_cldrVersion": "27.0.1",
>	          "_number": "$Revision: 11348 $"

**Comment by rxaviers** [comment]
I'm open to consider ideas and pull requests to optionally make the parser less permissive. Note JavaScript has similar behavior in what's regard extra characters tolerated in some situations: `parseInt('35NaN') // 35, same with parseFloat().`

Instead of the suggested fix, I prefer a more general solution. Something analogous to our date parser: [date/tokenizer-properties.js](https://github.com/jquery/globalize/blob/master/src/date/tokenizer-properties.js) and the [date/tokenizer.js](https://github.com/jquery/globalize/blob/master/src/date/tokenizer.js), which is what #353 is supposed to do.

Your help is very much appreciated. Closing this in favor of #353.

**Comment by robwade** [comment]
Thanks @rxaviers.

Happy to contribute if I can because I recently upgraded from 0.9 and am finding a few issues are catching me out.  The missing implementation of parseCurrency is another which #353 will resolve.

I've got workarounds in place which aren't compatible with #353 (and feel a little like dirty hacks).  Will try to make time in my schedule to help author a more permanent solution that is compatible with my validation helper method (below).

What's the best way to contribute?

```
isFloat: function (value, parseResults, options) {
	///<summary>Returns true if value is a float.</summary>
	///<param name="value">The value to test.</param>
	///<param name="parseResults">(optional) An object containing the parsed <c>value</c> and an <c>error</c> if parse failed.</param>
	///<param name="options">(optional) An object with:
	///		curr [String]: (no default) The currency the Number is in.
	///		errorMsgs [Object]: defaults to: {
	///			isNull: "Null or undefined value",
	///			isNaN: "Not a number or number string",
	///			tooPrecise: "Too many decimal places"
	///		}
	///		formatVal [Function]: If the value is valid, then this function (if specified) is called to reformat the value when setting parseResults.value.
	///		Globalize [Object]: The instance of Globalize to use (the window's Globalize object by default)
	///		gOpts [Object]: (no default) The Globalize options - see <c>Globalize.parseNumber</c> documentation.
	///		maxDecimalPlaces [Number]: The maximum number of decimal places.
	///</param>
```

**Comment by rxaviers** [comment]
Hey @robwade, I believe the best way to contribute is to start contributing :). The best first step to contribute a fix to this issue is to read #353 and let me know (preferably there) if you have any questions... Thanks and looking forward to hearing from you.
